### ISSUE TITLE
Barcode scanning not obvious working

### ISSUE THREAD

**Issue by christianlupus** [partkeepr/PartKeepr#1068]
[comment]: # (Hi and thanks for an issue. In order to help you best, please fill out the following information.)

I am willing to put labels on my parts. Ideally I would have an option to scan for parts using a barcode or the like. However, I would need to know what exactly I put into the barcodes before I start printing hundreds of labels just to see the barcode is not working.

I think there are multiple way in general: One is to [use the search field](https://github.com/partkeepr/PartKeepr/issues/316#issuecomment-19414654). The other is [described in the wiki](https://wiki.partkeepr.org/wiki/Barcode_Scanning/Configuration). The foremost has the problem that the search functionality for internal ids is non-functional at the moment (#924).

When trying to use the latter method, I do not get any useful results. I do not have a barcode reader that allows me to insert a hotkey automatically. So I need to manually press the hotkey then "type" the search string. Is this correct? I tried on the demo page to set the hotkey to `<Ctrl><Alt><Shift><I>` and added an action for key `S` in the table below. Saving and going back to the main parts view I do the following:
1. Press the Hotkey (aka `<Ctrl><Alt><Shift><I>`) and letting loose all 4 keys
2. Typing `s767` and Enter
I assume something should happen now and the part with number `767` should be highlighted or the like. 

Does this happen for you as well or am I misinterpreting the intended use of the function?

**Comment by dromer** [comment]
At the hackerspace we only use barcodes for storage locations, which is just a human readable string.
Our barcode-scanner behaves like a regular USB Human Input Device.

To search:
- click filter at the bottom of the main screen
- storage location input field
- scan the code
- find items

Similar for adding (storage location must already exist, but you can add it if it's not in the list yet).

**Comment by dromer** [comment]
It would be great to have hotkeys hooked straight into 'location search' or 'add part with this location'.

**Comment by christianlupus** [comment]
> Our barcode-scanner behaves like a regular USB Human Input Device.

So do mine.
I was just unsure, if I need to create a "special" set of barcodes in oder to make them use the search function automatically. In this case I would have to do this before starting to print them out :smirk:.

> It would be great to have hotkeys hooked straight into 'location search' or 'add part with this location'.

I think here it would be easier to have QR codes pointing to matching URLs. This would allow much more flexibility also to avoid conflicts with hotkeys from the browser (I had a hard time to find a free hotkey). But this imples #635, #1007 and others.

**Comment by christianlupus** [comment]
Yes, that is true. I have not (yet) thought it through completely. It was just a first guess.

But this was exactly my thinking that such a preamble might be needed and functional. But I think it is not. So I wil make simple labels. Maybe I will add a QR later if needed....

**Comment by ghost-from-the-past** [comment]
The actual implementation (with the special key) is not nice to have a keyboard free procedure
I described the actual procedure and some tips here
https://readthedocs.web.cern.ch/display/PARTK/Barcode+Readers
https://readthedocs.web.cern.ch/display/PARTK/barcode+reader+example

I believe that it is better to have a special sequence of chars like @@@@@ or ##### or ^^^^^, that are easy to print which could trigger the special behavior
https://readthedocs.web.cern.ch/display/PARTK/barcode+reading+method+2
for the moment I'm not very skilled in PHP,java script, but I'm learning and I'll try to implement it
Regards
Daniel
